pipeline/matching/core/analytics.py

```diff
@@ -0,0 +1,27 @@
+from pipeline.matching.core import MatchingEngine
+from pipeline.types import MatchingInput, MatchingInternalAnalytics
+
+
+class CollectInternalAnalytics:
+    def __init__(self, name, matching_engine):
+        self.name: str = name
+        self.matching_engine: MatchingEngine = matching_engine
+
+    def n_proposed_matches_per_user(
+        self, inp: MatchingInput
+    ) -> MatchingInternalAnalytics:
+
+        n_proposed_matches_per_user = []
+        for user in inp.users:
+            match_count = 0
+            for generator in self.matching_engine.generators:
```

Comment:
What is the runtime of this code? Can it be improved?

Reply:
Chris and I discussed how to improve the runtime! Chris was thinking about the case where some users get no matches and how we want to include a zero count for them, but that can be done more efficiently as well. Good job improving the performance of this, Chris!